spec/helpers.lua

```diff
@@ -49,7 +110,19 @@ function resty_http_proxy_mt:send(opts)
     opts.query = nil
   end
 
-  return self:request(opts)
+  local res, err = self:request(opts)
+  if res then
+    -- wrap the read_body() so it caches the result and can be called multiple times
+    local reader = res.read_body
+    res.read_body = function(self)
+      if (not self._cached_body) and (not self._cached_error) then
```

Comment:
???

If we only use the parenthesis is cases where not using them creates a bug, then it's no much use to prevent bugs as we're not coding for bugs, no? That's like 'we do defensive programming' only if we have a bug? That's bugfixing, not defensive programming. Defensive programming is reducing the possibility of bugs, on all levels of; technical, human, code writing, and code maintaining levels.

eg. using parenthesis would probably have prevented this `or nil` masterpiece; https://github.com/Mashape/kong/blob/0.8.0/kong/core/plugins_iterator.lua#L66